src/gui/guiChatConsole.cpp

```diff
@@ -528,96 +529,97 @@ bool GUIChatConsole::OnEvent(const SEvent& event)
 				scope);
 			return true;
 		}
-		else if(event.KeyInput.Key == KEY_KEY_A && event.KeyInput.Control)
-		{
-			// Ctrl-A pressed
-			// Select all text
-			prompt.cursorOperation(
-				ChatPrompt::CURSOROP_SELECT,
-				ChatPrompt::CURSOROP_DIR_LEFT, // Ignored
-				ChatPrompt::CURSOROP_SCOPE_LINE);
+		case KEY_TAB: {
+			// Tab or Shift-Tab pressed
+			// Nick completion
+			std::list<std::string> names = m_client->getConnectedPlayerNames();
+			bool backwards = event.KeyInput.Shift;
+			prompt.nickCompletion(names, backwards);
 			return true;
 		}
-		else if(event.KeyInput.Key == KEY_KEY_C && event.KeyInput.Control)
-		{
-			// Ctrl-C pressed
-			// Copy text to clipboard
-			if (prompt.getCursorLength() <= 0)
-				return true;
-			std::wstring wselected = prompt.getSelection();
-			std::string selected = wide_to_utf8(wselected);
-			Environment->getOSOperator()->copyToClipboard(selected.c_str());
-			return true;
+		default: break;
 		}
-		else if(event.KeyInput.Key == KEY_KEY_V && event.KeyInput.Control)
-		{
-			// Ctrl-V pressed
-			// paste text from clipboard
-			if (prompt.getCursorLength() > 0) {
-				// Delete selected section of text
+
+		if (event.KeyInput.Control) {
+			switch (key_code) {
```

Comment:
Note to reviewers: Hide whitespace changes for a saner diff. Although the logic wasn't touched.